bindings/python/test/time/test_interval.py

```diff
@@ -2,137 +2,219 @@
+    return Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.HalfOpenLeft)
 
-################################################################################################################################################################
+@pytest.fixture
+def interval_1 (closed_interval: Interval) -> Interval:
 
-def test_interval_operators ():
+    return closed_interval
 
-    interval = Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.Closed)
+@pytest.fixture
+def interval_2 () -> Interval:
 
-    assert (interval == interval) is not None
-    assert (interval != interval) is not None
+    return Interval(Instant.J2000() - Duration.minutes(1.0), Instant.J2000(), Interval.Type.Closed)
 
-################################################################################################################################################################
+@pytest.fixture
+def interval_3 () -> Interval:
 
-def test_interval_is_defined ():
+    return Interval(Instant.J2000() - Duration.minutes(2.0), Instant.J2000() - Duration.minutes(1.0), Interval.Type.Closed)
 
-    interval = Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.Closed)
+@pytest.fixture
+def interval_4 () -> Interval:
 
-    assert interval.is_defined() is not None
+    return Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.Open)
 
-################################################################################################################################################################
+@pytest.fixture
+def interval (interval_1: Interval) -> Interval:
 
-def test_interval_is_degenerate ():
+    return interval_1
 
-    interval = Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.Closed)
+@pytest.fixture
+def instant_1 () -> Interval:
 
-    assert interval.is_degenerate() is not None
+    return Instant.J2000()
 
-################################################################################################################################################################
+@pytest.fixture
+def instant_2 () -> Interval:
 
-@pytest.mark.skip
-def test_interval_intersects ():
+    return Instant.J2000() + Duration.seconds(30.0)
 
-    interval = Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.Closed)
+@pytest.fixture
+def instant_3 () -> Interval:
 
-    assert interval.intersects(interval) is not None
+    return Instant.J2000() + Duration.minutes(3.0)
 
 ################################################################################################################################################################
 
-def test_interval_contains_instant ():
+class TestInterval:
 
-    interval = Interval(Instant.J2000(), Instant.J2000() + Duration.minutes(1.0), Interval.Type.Closed)
+    def test_interval_constructor (self):
```

Comment:
Detail: did we drop the tested class name in the test names already elsewhere (in this project)?

```cpp
class TestClass:
      def test_constructor (self):
```
instead of:
```cpp
class TestClass:
      def test_class_constructor (self):
```

If we're in an in-between state, let's realign everything later

Reply:
Yeah, we are in between (as you can see not everything as arg, not all the tests use fixtures), we are re-aligning as we go